modules/swagger-parser-v3/src/main/java/io/swagger/parser/v3/OpenAPIV3Parser.java

```diff
@@ -54,12 +60,22 @@ public SwaggerParseResult readContents(String swaggerAsString, List<Authorizatio
             else {
                 mapper = YAML_MAPPER;
             }
-            try {
-                OpenAPI api = mapper.readValue(swaggerAsString, OpenAPI.class);
-                result.setOpenAPI(api);
+            if(auth == null) {
+                auth = new ArrayList<>();
             }
-            catch (Exception e) {
-                result.setMessages(Arrays.asList(e.getMessage()));
+            if(options != null) {
+                if (options.isResolve()) {
+                    try {
+                        OpenAPIDeserializer deserializer = new OpenAPIDeserializer();
+                        JsonNode rootNode = mapper.readTree(swaggerAsString.getBytes());
+                        SwaggerParseResult deserializeOpenAPI = deserializer.deserialize(rootNode);
+                        OpenAPIResolver resolver = new OpenAPIResolver(deserializeOpenAPI.getOpenAPI(), auth, null);
+                        result.setOpenAPI(resolver.resolve());
+
+                    } catch (Exception e) {
+                        result.setMessages(Arrays.asList(e.getMessage()));
+                    }
+                }
```

Comment:
Same with above, please use the deserializer if resolve is not set, or if options are null